### DISCUSSION TITLE
Is it possible to make extracted URLs from text fields machine translatable?

### DISCUSSION THREAD

**Discussion by leonh** [wagtail/wagtail-localize#414] [Ideas]
Custom machine translators have been a great time saver they are a really useful feature however they ignore translation  of extracted url's
 
The machine_translate function in edit_translation only attempts to translate stringsegment , links are extracted to  overrideableSegments  

context:
https://github.com/wagtail/wagtail-localize/pull/296
https://github.com/wagtail/wagtail-localize/pull/296#issuecomment-754151854
https://github.com/wagtail/wagtail-localize/blob/116582ccd39dae39d4c5dc0a34acbdb3b6984f5a/wagtail_localize/views/edit_translation.py#L949

Would it be possible to enable optional machinetranslation of extracted links?

**Comment by kaedroho** [comment]
This, obviously, wouldn't be possible with the current machine translators since Google Translate / DeepL don't know how to translate an arbitrary URL.

We could probably allow someone to hook in some code somewhere to allow them to translate URLs for specific domains that they know how to translate.

**Reply by leonh** [reply]
thanks for responding .

Yes, you're right it wouldn't work with translation services like  eg DeepL. 

We have used a very basic custom machine translate class to help content authors translate various english versions of a site UK, US, NZ AU the set of translations is known to us , running machine translate is a big time saver. 
However our extracted landing page urls need translation . 

For example we have internal links in content that need to be translated. like this:

/en-gb/rent/flat
to
/en-us/rent/apartment  

some optional ability to deal with overrideableSegments in machine translate would be very helpful.
